core/src/main/java/com/linecorp/armeria/server/http/auth/BasicToken.java

```diff
@@ -0,0 +1,93 @@
+package com.linecorp.armeria.server.http.auth;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.Base64.Encoder;
+import java.util.Objects;
+
+/**
+ * The bearer token of <a href="https://en.wikipedia.org/wiki/Basic_access_authentication">HTTP Basic access authentication</a>.
+ * */
+public final class BasicToken {
+
+    private static final Encoder BASE64_ENCODER = Base64.getEncoder();
+    private static final Decoder BASE64_DECODER = Base64.getDecoder();
+
+    /**
+     * Creates a new {@link BasicToken} from given {@code username} and {@code password}.
+     * */
+    public static BasicToken of(String username, String password) {
+        return new BasicToken(username, password);
+    }
+
+    /**
+     * Creates a new {@link BasicToken} from the given base64-encoded string.
+     * */
+    public static BasicToken decode(String base64) {
+        String credential = new String(BASE64_DECODER.decode(base64), StandardCharsets.UTF_8);
+        int sep = credential.indexOf(':');
+        if (sep == -1) {
+            throw new IllegalArgumentException("Invalid credential: " + credential);
+        }
+        String username = credential.substring(0, sep);
+        String password = credential.substring(sep + 1);
+        return of(username, password);
+    }
+
+    private final String username;
+    private final String password;
+
+    private BasicToken(String username, String password) {
+        requireNonNull(username, "username");
+        requireNonNull(password, "password");
+
+        this.username = username;
```

Comment:
Wow, I see. I will apply this for all requireNonNull invocations by next update.